### PR TITLE
Add improvements to near-field masking to the main branch

### PR DESCRIPTION
**Pull request by psheehan**
@jjtobin, experimenting with whether I can use git to merge the updates to near-field masking into your main, which would hopefully make life a lot easier down the road when I merge the rest of the changes. There are still a few commits I need to pull in, I'm mostly opening this as it will make it easier to visualize the changes and make sure it is getting what we want, so leave open for the time being.  There appear to be conflicts, so its not yet possible to merge anyways....

### CODE REVIEW

**Comment by psheehan** [comment]
@jjtobin, ok, I've fixed the conflicts and cherry-picked in a few additional commits that are relevant. At this point the update_nfsnr_merge_jtobin_main branch should include these new features:

- Improved near-field mask generation
- Use of nfrms_multiplier to adjust automasking thresholds

As a note, this is not *the* most up-to-date version of estimate_near_field_SNR; there were a few changes that were made to better deal with mosaics, but as PL2023 won't include mosaics, I did not include those here.

Could you take a look through the diff: https://github.com/jjtobin/auto_selfcal/pull/15/files and make sure that this includes what you are expecting and nothing you aren't?

I'm going to test the branch out on a few datasets to make sure it is working as expected, and might be worthwhile for you to do the same if you have a chance.

File: selfcal_helpers.py [hunk 1]
```diff
@@ -1619,14 +1648,19 @@ def get_max_uvdist(vislist,bands,band_properties):
...
+      min_baseline=np.min(all_baselines)
+      baseline_5=numpy.percentile(all_baselines,5.0)
       baseline_75=numpy.percentile(all_baselines,75.0)
       baseline_median=numpy.percentile(all_baselines,50.0)
       for vis in vislist:
          meanlam=3.0e8/band_properties[vis][band]['meanfreq']
          max_uv_dist=max_baseline # leave maxuv in meters like the other uv entries /meanlam/1000.0
+         min_uv_dist=min_baseline
          band_properties[vis][band]['maxuv']=max_uv_dist
+         band_properties[vis][band]['minuv']=max_uv_dist
```
Comment (jjtobin) [inline selfcal_helpers.py:1660]: should this be min_uv_dist and not max_uv_dist?

Comment (psheehan) [reply selfcal_helpers.py:1660]: Indeed it should, though I don't think it is actually used anywhere. Still, I'll fix that up

Comment (psheehan) [reply selfcal_helpers.py]: Should be fixed now

**Pull request merged**